be/src/exec/vectorized/sorting/sorting.h

```diff
@@ -50,5 +52,92 @@ void build_tie_for_column(const ColumnPtr column, Tie* tie);
 // Append rows from permutation
 void append_by_permutation(Column* dst, const Columns& columns, const Permutation& perm);
 void append_by_permutation(Chunk* dst, const std::vector<ChunkPtr>& chunks, const Permutation& perm);
+void append_by_permutation(Chunk* dst, const std::vector<const Chunk*>& chunks, const Permutation& perm);
+
+struct SortDesc {
+    int sort_order;
```

Comment:
why not use bool type for sort_order and null_first?

Reply:
actually bool type is more complicated: 

```c++
    for (size_t i = 0; i < col_num; ++i) {
        _sort_order_flag[i] = (*is_asc)[i] ? 1 : -1;
        if ((*is_asc)[i]) {
            _null_first_flag[i] = (*is_null_first)[i] ? -1 : 1;
        } else {
            _null_first_flag[i] = (*is_null_first)[i] ? 1 : -1;
        }
    }
```